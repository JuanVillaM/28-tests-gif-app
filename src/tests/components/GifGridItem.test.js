import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Tests in <GifGridItem />', () => {

    const title = 'title';
    const url = 'https://url';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    test('Should show the component correctly.', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Should have the <p> with the title.', () => {
        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );
    });
    
    test('Image should have src and alt.', () => {
        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test('Should have the class animate__fadeIn.', () => {
        const div = wrapper.find('div');
        const className = div.props('className');

        expect( className.includes('animate__fadeIn') ).toBe( true );
    })
    
});
