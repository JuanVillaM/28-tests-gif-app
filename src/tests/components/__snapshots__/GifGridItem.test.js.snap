// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem /> Should show the component correctly. 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    src="https://url"
  />
  <p>
     
    title
     
  </p>
</div>
`;
